### PR TITLE
Expand SRP Task 01

### PR DESCRIPTION
**Pull request by KyleKing**
I simplified the first task to just getting familiar with code reload and how the sprite paths work. There is an example solution they can uncomment to make it even easier

I also added a lot of explanation and examples that hopefully better illustrate the concept as well

### CODE REVIEW

File: game/tasks/task01_S_player.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+from ..core.registration import Register, SpriteRegister
+from ..core.views.rpg_sprites import CharacterSprite, PlayerSprite
+
+"""
+
+Each task can be reloaded when you make code changes with either `Ctrl R` or `CMD R`!
+
+To play the game with reload, both `SOURCE_NAME` and `load_sprites` need to be
+implemented. If these get removed, you will see an error when reloading. After fixing,
+you can reload without needing to quit.
```
Comment (dimmerex) [inline game/tasks/task01_S_player.py:27]: I think there's a chance the user might misunderstand and think they need to implement `SOURCE_NAME` and `load_sprites`.

Maybe we can rephrase it to something like this?
```
If reload is not working, you may have accidentally removed one of the following from the file:
- `SOURCE_NAME` variable
- `load_sprites` function
```

File: game/tasks/task01_S_player.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+class SpriteRegister2:
+    """Just a Register method."""
+
+    def register_sprite(self, sprite: arcade.Sprite, sprite_register: SpriteRegister) -> None:
+        register = Register(sprite=sprite, source=SOURCE_NAME)
+        sprite_register.register_sprite(register)
+
+
+def load_sprites_2(sprite_register: SpriteRegister) -> None:
```
Comment (dimmerex) [inline game/tasks/task01_S_player.py:167]: What's the purpose of adding numbers at the end of `load_sprite`, `Sprite` and `SpriteRegister`?

**Review by dimmerex — APPROVED** [review]
Added some minor comments, but otherwise looks good!

File: game/tasks/task01_S_player.py (hunk 1 above)
Comment (KyleKing) [reply game/tasks/task01_S_player.py:27]: That's a good point! I think this might be a little more clear, but we can fix it up in future commits as well

```suggestion
Avoid making any changes to `SOURCE_NAME` or the function signature to `load_sprites`
because those are required for reload, but you can change the content of the
`load_sprites` function. On reload, the game will attempt to identify any errors and
recommended fixes, so you can often fix and reload without guitting!
```

File: game/tasks/task01_S_player.py (hunk 2 above)
Comment (KyleKing) [reply game/tasks/task01_S_player.py:168]: The numbers are to avoid shadowing/name collision, but I can rework those to be underscores and add a note
